examples/employees-0.md

```diff
@@ -0,0 +1,63 @@
+```elm
+# Task: rank the employee titles according to the average salary for each department.
+
+# My solution:
+# - for each employee, find their average salary,
+# - join employees with their departments and titles (duplicating employees for each of their titles and departments)
+# - group by department and title, aggregating average salary
+# - join with department to get department name
+
+from employees
+join side:left salaries [emp_no]
+aggregate by:[employees.emp_no] [
+  emp_salary: average salary
+]
+join side:left titles [emp_no]
+join dept_emp [emp_no]
+aggregate by:[dept_emp.dept_no, titles.title] [
+  avg_salary: average emp_salary
+]
+join side:left departments [dept_no]
+select [dept_name, title, avg_salary]
+
+# Note that this solution contains a common pitfall in the first few lines: 
```

Comment:
I hear your point about PRQL handling imperfect queries well — I'm just not sure we want the first example to not be the most efficient way of representing the goal — does it communicate the strength of PRQL as well as a more efficient query?

But overall this PR is excellent, I don't want to be dogmatic on the small things — I'm very happy to merge and we can reflect and iterate on these in the future.